app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -1049,8 +1065,74 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
         private const val TAG = "MainActivity" // logging tag
         private const val DRAWER_ITEM_ADD_ACCOUNT: Long = -13
         private const val DRAWER_ITEM_ANNOUNCEMENTS: Long = 14
-        const val REDIRECT_URL = "redirectUrl"
-        const val OPEN_DRAFTS = "draft"
+        private const val REDIRECT_URL = "redirectUrl"
+        private const val OPEN_DRAFTS = "draft"
+        private const val ACCOUNT_ID = "accountId"
+        private const val COMPOSE_OPTIONS = "composeOptions"
+        private const val NOTIFICATION_TYPE = "notificationType"
+        private const val NOTIFICATION_TAG = "notificationTag"
+        private const val NOTIFICATION_ID = "notificationId"
+
+        /**
+         * switches the active account to the provided accountId and then stays on MainActivity
+         */
+        @JvmStatic
+        fun accountSwitchIntent(context: Context, accountId: Long): Intent {
+            return Intent(context, MainActivity::class.java).apply {
+                putExtra(ACCOUNT_ID, accountId)
+            }
+        }
+
+        /**
+         * switches the active account to the accountId and takes the user to correct place according to the notification they clicked
+         */
+        @JvmStatic
+        fun openNotificationIntent(context: Context, type: Notification.Type, accountId: Long): Intent {
+            return accountSwitchIntent(context, accountId).apply {
+                putExtra(NOTIFICATION_TYPE, type)
+            }
+        }
+
+        /**
+         * switches the active account to the accountId and then opens the ComposeActivity with the provided options
+         * @param notificationId optional id of the notification that should be cancelled when this intent is opened
+         * @param notificationTag optional tag of the notification that should be cancelled when this intent is opened
+         */
+        @JvmStatic
+        fun composeIntent(
+            context: Context,
+            options: ComposeActivity.ComposeOptions,
+            accountId: Long = -1,
+            notificationTag: String? = null,
+            notificationId: Int = -1
+        ): Intent {
+            return accountSwitchIntent(context, accountId).apply {
+                action = Intent.ACTION_SEND // so it can be opened via shortcuts
+                putExtra(COMPOSE_OPTIONS, options)
+                putExtra(NOTIFICATION_TAG, notificationTag)
+                putExtra(NOTIFICATION_ID, notificationId)
```

Comment:
Should these be conditional on `notificationTag` being non-null and `notificationId` not being `-1`?